src/openrct2/object/RideObject.cpp

```diff
@@ -410,6 +403,33 @@ void RideObject::SetRepositoryItem(ObjectRepositoryItem * item) const
         flags |= ORI_RIDE_FLAG_SEPARATE;
     }
     item->RideFlags = flags;
+
+
+
+    // Find the first non-null ride type, to be used when checking the ride group
+    uint8 rideTypeIndex = 0;
+    while (_legacyType.ride_type[rideTypeIndex] == RIDE_TYPE_NULL && rideTypeIndex < 3)
```

Comment:
You need to move the `rideTypeIndex < 3` check to the beginning, otherwise it will access element 3 (out of range) of `ride_type` still.